packages/monocdk-experiment/gen.js

```diff
@@ -0,0 +1,99 @@
+// generates the mono-cdk module by copying @aws-cdk/lib/** to src/
+// and rewriting the interdepedent "import" statements.
+const fs = require('fs-extra');
+const path = require('path');
+const glob = require('glob');
+
+async function main() {
+  const srcdir = path.resolve('src');
+  const reexports = [];
+
+  await fs.remove(srcdir);
+  await fs.mkdir(srcdir);
+
+  const root = path.resolve(__dirname, '..', '@aws-cdk');
+  const modules = await fs.readdir(root);
+
+  for (const dir of modules) {
+    const moduledir = path.resolve(root, dir);
+    const meta = JSON.parse(await fs.readFile(path.join(moduledir, 'package.json'), 'utf-8'));
+
+    if (meta.deprecated) {
+      console.error(`skipping deprecated ${meta.name}`);
+      continue;
+    }
+
+    if (!meta.jsii) {
+      console.error(`skipping non-jsii ${meta.name}`);
+      continue;
+    }
+
+    // check if moduledir includes any directory other than "lib" and "test"
+    const subdirs = [];
+    const allowed = [ 'lib', 'test', 'node_modules', 'scripts', 'build-tools' ];
+    for (const file of await fs.readdir(moduledir)) {
+      if (allowed.includes(file)) {
+        continue;
+      }
+      if ((await fs.stat(path.join(moduledir, file))).isDirectory()) {
+        subdirs.push(file);
+      }
+    }
+
+    if (subdirs.length > 0) {
+      console.error(`WARNING: ${moduledir} includes a directory that is not one of [${allowed.join(',')}]: [${subdirs.join(',')}]`);
+    }
+
+    const basename = path.basename(moduledir);
+    const source = `${moduledir}/lib`;
+    const target = `${srcdir}/${basename}/lib`;
+    await fs.copy(source, target);
+
+    await fs.writeFile(path.join(path.dirname(target), 'index.ts'), `export * from './lib'\n`);
+
+    const namespace = basename.replace(/-/g, '_');
+    reexports.push(`import * as ${namespace} from './${basename}/lib'; export { ${namespace} };`)
+  }
+
+  await fs.writeFile(path.join(srcdir, 'index.ts'), reexports.join('\n'));
+
+  const sourceFiles = await findSources(srcdir);
+  for (const source of sourceFiles) {
+    await rewriteImports(srcdir, source);
+  }
+}
+
+async function findSources(srcdir) {
+  return new Promise((ok, ko) => glob('**/*.ts', { cwd: srcdir }, (err, results) => {
```

Comment:
Just checking: we're confident we don't have any packaged lambda function handler code that uses javascript, yes?

Reply:
We might have but this is first approximation

Reply:
We have at least 3:
https://github.com/aws/aws-cdk/blob/b707782ecfd49ec7e907ec4331024ce23ae2fbdb/packages/cdk-dasm/jest.config.js
https://github.com/aws/aws-cdk/blob/ec367c8733356187e2b89d757dc69125d99343e4/packages/%40aws-cdk/aws-dynamodb-global/lambda-packages/aws-global-table-coordinator/lib/index.js
https://github.com/aws/aws-cdk/blob/db77bfecb98ffdcb092cf74bc7d79e6880a73b69/packages/%40aws-cdk/aws-certificatemanager/lambda-packages/dns_validated_certificate_handler/lib/index.js

Reply:
Definitely, this will be handled in a subsequent commit.